packages/datadog-instrumentations/src/prisma.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const { isPromise } = require('util/types')
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const primsaClientStartCH = channel('apm:prisma:client:start')
+const primsaClientFinishCH = channel('apm:prisma:client:finish')
+const prismaClientErrorCH = channel('apm:prisma:client:error')
+
+const prismaEngineStart = channel('apm:prisma:engine:start')
+
+const allowedClientSpanOperations = [
+  'operation',
+  'serialize',
+  'transaction'
+]
+class TracingHelper {
+  isEnabled () {
+    return true
+  }
+
+  // needs a sampled tracecontext to generate engine spans
+  getTraceParent (context) {
+    return '00-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-bbbbbbbbbbbbbbbb-01' // valid sampled traceparent
+  }
+
+  dispatchEngineSpans (spans) {
+    const roots = spans.filter((span) => span.parentId === null)
+    for (const root of roots) {
+      prismaEngineStart.publish({ engineSpan: root, allEngineSpans: spans })
+    }
+  }
+
+  getActiveContext () {}
+
+  runInChildSpan (options, callback) {
+    if (typeof options === 'string') {
+      options = {
+        name: options
+      }
+    }
+
+    if (allowedClientSpanOperations.includes(options.name)) {
+      const asyncResource = new AsyncResource('bound-anonymous-fn')
+      const ctx = {
+        resourceName: options.name,
+        attributes: options.attributes || {}
+      }
+
+      return asyncResource.runInAsyncScope(() => {
+        if (!primsaClientStartCH.hasSubscribers) {
+          return callback.apply(this, ctx)
+        }
+        primsaClientStartCH.publish(ctx)
+        try {
+          const result = callback.apply(this, ctx)
+          if (isPromise(result)) {
+            result.catch((error) => {
+              prismaClientErrorCH.publish({ error, ctx })
+              throw error
+            })
+              .finally(() => {
+                if (primsaClientFinishCH.hasSubscribers) {
+                  primsaClientFinishCH.publish()
+                }
+              })
+          } else {
+            if (primsaClientFinishCH.hasSubscribers) {
+              primsaClientFinishCH.publish()
+            }
+          }
+          return result
+        } catch (error) {
+          prismaClientErrorCH.publish({ error, ctx })
+          throw error
+        }
+      })
+    }
+    return callback()
+  }
+}
+
+// Tracing support GA in  v6.1.0+
+// https://github.com/prisma/prisma/releases/tag/6.1.0
+addHook({ name: '@prisma/client', versions: ['>=6.1.0'] }, prisma => {
+  const tracingHelper = new TracingHelper()
+  global.PRISMA_INSTRUMENTATION = {
+    helper: tracingHelper
+  }
+
+  global.V0_PRISMA_INSTRUMENTATION = {
+    helper: tracingHelper
+  }
```

Comment:
Def gonna need comments

Reply:
Comments and links have been added to explain the process